doc/user/content/transform-data/patterns/window-functions.md

```diff
@@ -103,16 +107,27 @@ INSERT INTO measurements VALUES
 
 You can compute the differences between consecutive measurements using `LAG()`:
 ```mzsql
-SELECT time, value - LAG(value) OVER (ORDER BY time)
-FROM measurements;
+SELECT time, value - LAG(value) OVER (ORDER BY time) as diff_w_prev
+FROM measurements
+ORDER BY time;
 ```
 
 For better performance, you can rewrite this query using an equi-join:
 
 ```mzsql
-SELECT m2.time, m2.value - m1.value
+SELECT m2.time, m2.value - m1.value as diff_w_prev
 FROM measurements m1, measurements m2
-WHERE m2.time = m1.time + INTERVAL '1' MINUTE;
+WHERE m2.time = m1.time + INTERVAL '1' MINUTE
+ORDER BY m2.time;
 ```
 
-Note that these queries differ in whether they include the first timestamp (with a `NULL` difference). Using a `LEFT JOIN` would make the outputs match exactly.
+Note that these queries differ in whether they include the first timestamp (with
+a `NULL` difference).  A `RIGHT JOIN` would make the outputs match exactly.
```

Comment:
I changed the Left Join to a Right Join as I think the Left join was a typo?

Reply:
That's right. It'd be a `LEFT JOIN` only if you changed the order of the tables in the `FROM` clause, which...doesn't make sense, anyways!

Reply:
Similarly, I'd write the previous query as:

```sql
SELECT m1.time, m1.value - m2.value
FROM measurements m1, measurements m2
WHERE m1.time = m2.time + INTERVAL '1' MINUTE
ORDER BY m1.time;
```

Reply:
Oh, great minds think alike.  To keep the changes small, I didn't change it ... but I'm with you -- it does look weird.  I'll patch it.